embedded-graphics/src/primitives/polyline.rs

```diff
@@ -0,0 +1,402 @@
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.translate += by;
+
+        self
+    }
+}
+
+/// An iterator over all pixel positions on the polyline
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct Points<'a> {
+    vertices: &'a [Point],
+    translate: Point,
+    segment_iter: line::Points,
+}
+
+impl<'a> Points<'a> {
+    fn new<'b>(polyline: &'b Polyline<'a>) -> Self
+    where
+        'a: 'b,
+    {
+        polyline
+            .vertices
+            .split_first()
+            .and_then(|(start, rest)| {
+                // Polyline is 2 or more vertices long, return an iterator for it
+                rest.get(0).map(|end| Points {
+                    vertices: rest,
+                    translate: polyline.translate,
+                    segment_iter: Line::new(*start + polyline.translate, *end + polyline.translate)
+                        .points(),
+                })
+            })
+            .unwrap_or_else(||
+                // Polyline is less than 2 vertices long. Return a dummy iterator that will short
+                // circuit due to `stop: true`
+                Points {
+                    vertices: &[],
+                    translate: Point::zero(),
+                    segment_iter: Line::new(Point::zero(), Point::zero()).points()
+                })
+    }
+}
+
+impl<'a> Iterator for Points<'a> {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Some(p) = self.segment_iter.next() {
+            Some(p)
+        } else {
+            let (start, rest) = self.vertices.split_first()?;
+            let end = rest.get(0)?;
+
+            self.vertices = rest;
+
+            self.segment_iter = Line::new(*start + self.translate, *end + self.translate).points();
+
+            // Skip first point of next line, otherwise we overlap with the previous line
+            self.nth(1)
+        }
+    }
+}
+
+impl<'a, C> IntoIterator for &'a Styled<Polyline<'a>, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledPolylineIterator<'a, C>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StyledPolylineIterator {
+            stroke_width: self.style.stroke_width,
+            stroke_color: self.style.stroke_color,
+            line_iter: self.primitive.points(),
+        }
+    }
+}
+
+/// Pixel iterator for each pixel in the line
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct StyledPolylineIterator<'a, C>
+where
+    C: PixelColor,
+{
+    stroke_width: u32,
+    stroke_color: Option<C>,
+    line_iter: Points<'a>,
+}
+
+impl<'a, C: PixelColor> Iterator for StyledPolylineIterator<'a, C> {
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // Break if stroke width is zero
+        if self.stroke_width == 0 {
+            return None;
+        }
```

Comment:
Can this be handled outside the `next` function, by using a dummy iter for this unlikely case?

Reply:
I've moved it out to `into_iter` by kludging the stroke colour based on width. Can you think of a better way?

Reply:
I think that's OK, because we need to check for `stroke_color != None` in the `next` function anyway. It would be nice if we could also get rid of the `let stroke_color = stroke_color?`, but I don't think that's possible.

Reply:
Certainly possible but would probably lead to messier code. I'll leave it as is for now. Perf-wise, I imagine `self.line_iter().next()` has a lot more overhead than the stroke width check.

Reply:
Sure it's fine, the comment was meant for iterators in general. But creating an empty dummy iterator wouldn't be too messy.
Changing `pixel_color: Option<C>` to `pixel_color: C`  isn't possible, because you can't create a dummy `stroke_color` for any `C: PixelColor` when creating a dummy iter.